brush-core/src/shell.rs

```diff
@@ -1268,6 +1268,30 @@ impl Shell {
         self.env.get(name).map(|(_, var)| var)
     }
 
+    /// Tries to set a global variable in the shell's environment.
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The name of the variable to add.
+    /// * `var` - The variable contents to add.
+    pub fn set_env_global(&mut self, name: &str, var: ShellVariable) -> Result<(), error::Error> {
+        self.env.set_global(name, var)
+    }
+
+    /// Register a builtin to the shell's environment.
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The in-shell name of the builtin.
+    /// * `registration` - The registration handle for the builtin.
+    pub fn register_builtin<S: Into<String>>(
+        &mut self,
+        name: S,
+        registration: builtins::Registration,
+    ) {
+        self.builtins.insert(name.into(), registration.special());
```

Comment:
I hadn't noticed it before, but I believe the tests are failing because of the `.special()` call. Unless you were intentionally looking for a way to allow registering [*special* built-ins](https://www.gnu.org/software/bash/manual/html_node/Special-Builtins.html), then this should just pass through `registration` without marking it special.